src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,653 @@
+            if (venueShares != 0) {
+                uint256 venueProfitRatio = profitToVconRatio[venue];
+
+                uint256 vconEarnedPerShare = (Constants.ETH_GRANULARITY *
+                    venueProfit.toUint256() *
+                    venueProfitRatio) / venueShares;
+
+                /// gain scenario
+                venueLastRecordedVconSharePrice[venue] += vconEarnedPerShare
+                    .toUint128();
+            }
+        } else {
+            /// share price is 0, and requires initialization
+            venueLastRecordedVconSharePrice[venue] = Constants
+                .ETH_GRANULARITY
+                .toUint128();
+        }
+
+        /// update the venue's profit index
+        venueLastRecordedProfit[venue] = lastRecordedProfit.toUint128();
+
+        emit VenueIndexUpdated(venue, block.timestamp, lastRecordedProfit);
+    }
+
+    /// ---------- Governor-Only Permissioned API ----------
+
+    /// @notice governor only function to set the profit to VCON ratio
+    /// this function will not be callable if an underlying venue took a loss as `_accrue()`
+    /// will revert and no users will be able to withdraw their VCON.
+    function setProfitToVconRatio(
+        address venue,
+        uint256 newProfitToVconRatio
+    ) external onlyGovernor globalLock(1) {
+        /// lock to level 1 so that accrue can succeed
+        require(pcvOracle().isVenue(venue), "MarketGovernance: invalid venue");
+        _accrue(venue); /// ensure users receive all rewards from the old rate
+
+        uint256 oldProfitToVconRatio = profitToVconRatio[venue];
+        profitToVconRatio[venue] = newProfitToVconRatio;
+
+        emit ProfitToVconRatioUpdated(
+            venue,
+            oldProfitToVconRatio,
+            newProfitToVconRatio
+        );
+    }
+
+    /// @notice governor only function to set the PCV Router for market governance rebalances
+    function setPCVRouter(address newPcvRouter) external onlyGovernor {
+        address oldPcvRouter = pcvRouter;
+        pcvRouter = newPcvRouter;
+
+        emit PCVRouterUpdated(oldPcvRouter, newPcvRouter);
+    }
+
+    /// @notice governor only function to set a venue for market governance withdrawals in a given denomination
+    /// @param token underlying token for venue to handle
+    /// @param venue where funds of denomination `token` will be withdrawn to
+    /// @dev venue must be a whitelisted PCV Deposit,
+    /// and the venue's underlying token must match the token passed
+    function setUnderlyingTokenHoldingDeposit(
+        address token,
+        address venue
+    ) external onlyGovernor {
+        require(pcvOracle().isVenue(venue), "MarketGovernance: invalid venue");
+        require(
+            IPCVDepositV2(venue).balanceReportedIn() == token,
+            "MarketGovernance: underlying mismatch"
+        );
+
+        underlyingTokenToHoldingDeposit[token] = venue;
+
+        emit UnderlyingTokenDepositUpdated(token, venue);
+    }
```

Comment:
probably better fit in the pcvguardian, but this can be a refactor in a separate PR